dvc/repo/checkout.py

```diff
@@ -34,36 +36,37 @@ def _checkout(
 ):
     from dvc.stage import StageFileDoesNotExistError, StageFileBadNameError
 
-    stages = set()
-
     if not targets:
         targets = [None]
         _cleanup_unused_links(self)
 
+    pairs = set()
```

Comment:
Why don't we do `set()` in `Repo.used_cache()`? Does `.collect_granular()` return dups sometimes anyway? Or the reason is that targets might be duplicated?

Reply:
Targets might be duplicated, correct.

Reply:
Why don't we duplicate targets before correction then?

Reply:
@Suor because it is not trivial. E.g. `dir/stage.dvc` and `-R dir`. Or `--with-deps` for two stages of the same pipeline.